upgrade_tests/test_scenarios.py

```diff
@@ -286,3 +286,246 @@ def test_post_user_scenario_bug_1429201(self):
         self.assertIsNotNone(container_ids)
         self.assertIn('Error', result_fail.values()[0])
         self.assertIn('Complete', result_pass.values()[0])
+
+
+class Scenario_capsule_sync(TestCase):
+    """The test class contains pre-upgrade and post-upgrade scenarios to test if
+    package added to satellite preupgrade is synced to capsule post upgrade.
+
+    Test Steps:
+
+    1. Before Satellite upgrade, Sync a repo/rpm in satellite.
+    2. Upgrade satellite/capsule.
+    3. Run capsule sync post upgrade.
+    4. Check if the repo/rpm is been synced to capsule.
+
+    """
+    sat_host = get_satellite_host()
+    env.host_string = sat_host
+    env.user = 'root'
+    hammer.set_hammer_config()
+    repo_name = 'capsulesync_TestRepo'
+    repo_path = '/var/www/html/pub/preupgradeCapSync_repo/'
+    rpm_name = rpm1.split('/')[-1]
+    prod_name = 'Scenario_preUpgradeCapSync'
+    activation_key = os.environ.get(
+        'CAPSULE_AK', os.environ.get('RHEV_CAPSULE_AK'))
+    cv_name = 'Scenario_precapSync'
+    _, env_name = hammer.hammer_determine_cv_and_env_from_ak(
+        activation_key, '1')
+    org_id = '1'
+    repo_url = 'http://' + sat_host + '/pub/preupgradeCapSync_repo/'
+
+    def create_repo(self):
```

Comment:
True we can do that , What I thinking was better to raise new PR for that purpose, as it needs to change in earlier scenarios as well. What do you think ?